test/tutorials/lambda_theory.jl

```diff
@@ -186,7 +186,11 @@ x = Variable(:x)
 y = Variable(:y)
 ex = Apply(λ(:x, λ(:y, Apply(x, y))), y)
 g = EGraph{LambdaExpr,LambdaAnalysis}(ex)
-saturate!(g, λT)
+params = SaturationParams(
+  timer = false,
+  check_analysis = true
```

Comment:
```suggestion
  check_memo = true
  check_analysis = true
```

Let's be safe

Reply:
Activated check_memo as well. Note that in this branch check_memo still has a bug which means that it does currently not actually check all entries in the memoization map.  (see #238)